### PR TITLE
Query Jira API for maxResults param

### LINKED ISSUE #4789: Get the maxResults Jira parameter from Jira itself instead of hardcoding it
The maxResults parameter is used in the collector en the external server (IssueTracker class).

### PR DESCRIPTION
**Pull request by wkoot**
Closes #4789

### CODE REVIEW

File: components/collector/src/source_collectors/jira/issues.py [hunk 1]
```diff
@@ -115,3 +117,12 @@ def __get_sprint_names(cls, sprint_texts: list[str]) -> str:
         matches = [cls.SPRINT_NAME_RE.search(sprint_text) for sprint_text in sprint_texts]
         sprint_names = [match.group(1) for match in matches if match]
         return ", ".join(sorted(sprint_names))
+
+    async def _determine_max_results(self) -> int:
+        """Maximum number of issues to retrieve per page."""
+        if isinstance(self.MAX_RESULTS, int):
+            return self.MAX_RESULTS
```
Comment (fniessink) [inline components/collector/src/source_collectors/jira/issues.py]: Can't we make this a cached property so that `self.MAX_RESULTS` isn't needed anymore?

**Review by fniessink — CHANGES_REQUESTED** [review]

File: components/collector/src/source_collectors/jira/issues.py (hunk 1 above)
Comment (wkoot) [reply components/collector/src/source_collectors/jira/issues.py]: Well, it's a bit hairy due to the asyncio coroutine :( I originally wanted to fetch it in `__init__` but then I need to use a different url library, such as requests. I don't think we have an async `cached_property` version somewhere in `collector_utilities`, is this desirable?

File: components/collector/src/source_collectors/jira/issues.py [hunk 2]
```diff
@@ -115,3 +118,12 @@ def __get_sprint_names(cls, sprint_texts: list[str]) -> str:
         matches = [cls.SPRINT_NAME_RE.search(sprint_text) for sprint_text in sprint_texts]
         sprint_names = [match.group(1) for match in matches if match]
         return ", ".join(sorted(sprint_names))
+
+    async def _determine_max_results(self) -> int:
+        """Maximum number of issues to retrieve per page."""
+        if isinstance(self.max_results, int):  # NB: using cached_property is not doable on asyncio coroutine
+            return self.max_results  # type: ignore[unreachable]
```
Comment (fniessink) [inline components/collector/src/source_collectors/jira/issues.py]: Maybe the mypy warning can be gotten rid of by typing max_results as None or int?

```python
    max_results: int | None = None
```

**Review by fniessink — APPROVED** [review]

File: components/collector/src/source_collectors/jira/issues.py (hunk 2 above)
Comment (wkoot) [reply components/collector/src/source_collectors/jira/issues.py]: Excellent, seems to work.

**Pull request merged**